Data/Row/Internal.hs

```diff
@@ -399,6 +389,16 @@ type family MapR (f :: a -> b) (r :: [LT a]) :: [LT b] where
   MapR f '[] = '[]
   MapR f (l :-> v ': t) = l :-> f v ': MapR f t
 
+-- | Take two rows with the same labels, and apply the type operator from the
+-- first row to the type of the second.
+type family DotProduct (fs :: Row (* -> *)) (r :: Row *) :: Row * where
```

Comment:
I'd prefer to call this `Ap` or some such — `DotProduct` makes me think we're reducing the row somehow to a single element, but we're not.

Reply:
I wrote `DotProduct` when I really meant `HadamardProduct`, but `Ap` is a better name than either of those.